### DISCUSSION TITLE
Sending data back to frontend in POST method

### DISCUSSION THREAD

**Discussion by nbhector** [fastapi/fastapi#8798] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.post("/resume/", response_description="Get resume from user")
async def create_candidate(request: Request, file: UploadFile = File(...)) -> dict:
    parser = ResumeParser()
    candidate_info = {}
    user_resume = file.file
    text = parser.convert_pdf_to_txt(user_resume)
    resume_token = []
    resume_text = ''
    resume_token.append(text)

    for i in resume_token:
        resume_text += str(i)
```

### Description

* The async function receives a file-like object from the frontend.
* Document should be passed through a parser and the extracted information should be stored in a dict.
* I want the information to be sent back to the frontend within the same function, but not sure how since this is a POST method.
* Any ideas?

### Operating System

Linux

### FastAPI Version

0.61.1

### Python Version

3.6

**Comment by n8sty** [comment, marked as answer]
If your function `return`s _something_ that object will get sent back to whatever caller has made the POST request, in this case your frontend. You're not sending anything back now because you're missing a `return`.

**Comment by nbhector** [comment]
Here is the modified code:

```
@router.post("/resume/", response_description="Get resume from user")
async def create_candidate( request: Request, file: UploadFile = File(...), candidate: CandidateModel = Body(...)) -> dict:
    parser = ResumeParser()
    candidate_info = {}
    user_resume = file.file
    text = parser.convert_pdf_to_txt(user_resume)
    resume_token = []
    resume_text = ''
    resume_token.append(text)

    for i in resume_token:
        resume_text += str(i)
    candidate_name = parser.extract_name(resume_text)
    candidate_phone = parser.extract_mobile_number(resume_text)
    candidate_email = parser.extract_email(resume_text)
    # candidate_skills = parser.extract_skills(resume_text, skills_data)
    converted_uuid = json.dumps(uuid.uuid4(), cls=UUIDEncoder)

    candidate_info["name"] = candidate_name
    candidate_info["phone"] = candidate_phone
    candidate_info["email"] = candidate_email
    candidate_info["skills"] = dummyData
    candidates_info = jsonable_encoder(candidate_info)

    # request.app.mongodb["candidates"].insert_one(candidate_info)
    return JSONResponse(content=candidates_info)
```


Even with the return statement (and without making it a JSONResponse), I get a 422 error.  Do I need to feed in the Pydantic Model I created?

**Comment by David-Lor** [comment]
Your request body must be a valid CandidateModel object. But you're also asking for a file in the same request, and I'd say that is currently not possible (you're looking at a multipart request - unless something changed recently, fastapi cannot handle a multipart request with a Pydantic-model body).

About the 422 error, the HTTP response you get with it gives a detailed explanation on what/why failed.

**Comment by nbhector** [comment]
That makes sense.  What could I do then?  I'm thinking that I could store the data into the database and have my frontend fetch data by identifying the UUID.  However, it seems a bit complicated because the frontend needs to know the uuid right after the user uploads the file

**Comment by David-Lor** [comment]
Before advicing you about the multipart request with a JSON body, do you really need it? Because in your original code you did not specify it, and in your modified code you included it (`candidate: CandidateModel = Body(...)`), but is not used anywhere on the given code (maybe you will use it later?).

If your original question remains, you may want to define a Pydantic model that you'll use as a Response model class; then, you'd just have to return an instance of that class, with the data you want to return in the response body (FastAPI handles converting the class instance to JSON).
